### PR TITLE
#266 - Updated update operation request schema

### PR DESCRIPTION
**Pull request by thehenrytsai**

### CODE REVIEW

File: lib/core/versions/latest/UpdateOperation.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    this.operationData = operationData;
+  }
+
+  /**
+   * Parses the given buffer as a `UpdateOperation`.
+   */
+  public static async parse (operationBuffer: Buffer): Promise<UpdateOperation> {
+    const operationJsonString = operationBuffer.toString();
+    const operationObject = await JsonAsync.parse(operationJsonString);
+    const createOperation = await UpdateOperation.parseObject(operationObject, operationBuffer);
```
Comment (isaacJChen) [inline lib/core/versions/latest/UpdateOperation.ts]: updateOperation = ...

File: lib/core/versions/latest/UpdateOperation.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    return createOperation;
+  }
+
+  /**
+   * Parses the given operation object as a `CreateOperation`.
```
Comment (isaacJChen) [inline lib/core/versions/latest/UpdateOperation.ts]: as UpdateOperation

File: lib/core/versions/latest/util/Jws.ts [hunk 3]
```diff
@@ -22,14 +78,55 @@ export default class Jws {
...
 
+  /**
+   * Signs the given protected header and payload as a JWS.
+   *
+   * @param payload If the given payload is of string type, it is assumed to be encoded string;
+   *                else the object will be stringified and encoded.
+   */
+  public static async sign (
+    protectedHeader: any,
+    payload: any | string,
```
Comment (isaacJChen) [inline lib/core/versions/latest/util/Jws.ts]: Any covers string, unless you think it's better to be explicit.

File: tests/core/util/Jws.spec.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      const [, signingPrivateKey] = await Cryptography.generateKeyPairHex(signingKeyId, KeyUsage.signing);
+
+      const protectedHeader = {
+        kid: signingKeyId,
+        alg: true // Incorect type.
+      };
+
+      const payload = { anyProperty: 'anyValue' };
+
+      const jws = await Jws.sign(protectedHeader, payload, signingPrivateKey);
```
Comment (isaacJChen) [inline tests/core/util/Jws.spec.ts]: We should probably have a test for when payload is a string if we want to support it.

File: lib/core/versions/latest/DocumnetComposer.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import UpdateOperation from './UpdateOperation';
```
Comment (mudiali) [inline lib/core/versions/latest/DocumnetComposer.ts]: Typo in the filename.

File: lib/core/versions/latest/OperationProcessor.ts [hunk 6]
```diff
@@ -112,30 +121,25 @@ export default class OperationProcessor implements IOperationProcessor {
...
 
-    // The operation passes all checks, apply the patches.
-    AnchoredOperation.applyPatchesToDidDocument(didDocument, operation.patches!);
+    const resultingDocument = await this.documentComposer.applyUpdateOperation(operation, didResolutionModel.didDocument);
 
-    didResolutionModel.metadata!.lastOperationTransactionNumber = operation.transactionNumber;
-    didResolutionModel.metadata!.nextUpdateOtpHash = operation.nextUpdateOtpHash!;
+    didResolutionModel.didDocument = resultingDocument;
+    didResolutionModel.metadata!.lastOperationTransactionNumber = namedAnchoredOperationModel.transactionNumber;
+    didResolutionModel.metadata!.nextUpdateOtpHash = operation.operationData.nextUpdateOtpHash;
```
Comment (mudiali) [inline lib/core/versions/latest/OperationProcessor.ts]: I'm a bit confused as to how are you able to access the `operationData.nextUpdateOtpHash` without having to import it? Also, having the same name for the internal class (in both CreateOperation.ts and UpdateOperation.ts) was going to clash right?

File: lib/core/versions/latest/UpdateOperation.ts (hunk 1 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/UpdateOperation.ts]: Fixed.

File: lib/core/versions/latest/util/Jws.ts (hunk 3 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/util/Jws.ts]: Removed.

File: lib/core/versions/latest/UpdateOperation.ts (hunk 2 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/UpdateOperation.ts]: Fixed!

File: lib/core/versions/latest/DocumnetComposer.ts (hunk 5 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/DocumnetComposer.ts]: Thanks. Fixed.

File: tests/core/util/Jws.spec.ts (hunk 4 above)
Comment (thehenrytsai) [reply tests/core/util/Jws.spec.ts]: Added a note saying that this is mainly created for tests to create valid data, and the path is hit many times already.

File: lib/core/versions/latest/OperationProcessor.ts (hunk 6 above)
Comment (thehenrytsai) [reply lib/core/versions/latest/OperationProcessor.ts]: Short answer is that this is something that TypeScript handles gracefully.

**Pull request merged**